Fixed issue with "weird time" + updated locale + added Czech translation

vyskoczilova opened the pull request:
Hi, I have found your plugin and used it for my client. I had to fix few issues anyway so I would love to give you a helping hand. I hope that the thread is not dead. If you don't have time for it anymore, I can take over or be a co-author also at WordPress repo. If I have time I would love to add support for better settings (different times for different pickup locations made by shipping zones).

Fixes:
- it displays the very first pickup time possible (no more juggling with two hours later)
- if there is a possibility to order in advance (passed by your $num_days_allowed) it also lists the dates
- if "today" it's too late and the shop is closed, and you have $num_days_allowed > 1 it lists all the dates possible (without today's date)
- created translatable date names
- updated translation files
- added Czech translation

mattbanks commented:
@vyskoczilova I'm tied up this week with some projects, but I'll get this pulled down as soon as I can to check it out and get it merged in!

vyskoczilova commented:
@mattbanks Hi, great to hear that :)

mattbanks commented on public/class-local-pickup-time.php:
```diff
@@ -323,17 +326,54 @@ public function create_hour_options() {
 
-					$day_name = ( $i === 0 ) ? 'Today' : $current_day_name;
+							$option_key = $current_day_name . date( "_H_i", $tNow );
+							$option_value = $day_name . ' ' . date( "H:i", $tNow );
```
I think this date format should be made available as a filter, with `date( "g:i", $tNow )` as the default (what it was set before). I know that most US based users would prefer to use 12 hour time without a leading zero.

mattbanks commented:
Sorry for taking so long to get to this @vyskoczilova! 😢 

I had one comment that I left above. Let me know if you can make that change to add a filter for the drop down value.

vyskoczilova commented:
@mattbanks good morning :) thanks for answering no troubles, I will update that, together with other changes concerning newer WooCommerce version which I had to do also in meanwhile.

vyskoczilova commented:
@mattbanks Hi, I have finally uploaded my last commits:
- enable translatable pickup time (day's names) in email
- updated call for email and error message integration
- added select pickup time as required with empty option at the very begining of the options
- hided checkout payment if there are no times available in near future (limitted by user settings)
- settings for time are loaded from WP Options, which I found user more friendly than a filter
- updated the translations files

Does it look ok, to you? If you want to write it somewhere as a contributor, my WP name is same as here.

mattbanks commented:
looks good to me! merging this and will bump the version and add you to the contributors list! :)
